algorithms/nautilus/fuzzer/other_hierarflow_routines.cpp

```diff
@@ -43,20 +43,32 @@ NullableRef<HierarFlowCallee<void(void)>> FuzzLoop::operator()(void) {
 
 /**
  * @fn
- * @brief HierarFlow routine for SelectInput (select_input)
+ * @brief HierarFlow routine for SelectInput (select_input_and_switch)
  */
 RSelectInput SelectInput::operator()(void) {
-  std::optional<QueueItem> inp;
+  std::unique_ptr<QueueItem> inp;
+
+  auto& node = this->UnwrapCurrentLinkedNodeRef();
+  auto& succ_nodes = node.succ_nodes;
+
+  if (succ_nodes.size() != 2) {
+    /* We call either process_next_input or generate_input */
+    throw exceptions::wrong_hierarflow_usage(
+      "Invalid number of children for SelectInput",
+      __FILE__, __LINE__
+    );
+  }
 
   if (state.queue.IsEmpty()) {
-    inp = std::nullopt;
+    /* If queue is empty, generate a new testcase */
+    (*succ_nodes[1])(inp); // generate_input (inp is invalid here)
+
   } else {
-    // TODO: Use lock when multi-threaded
-    inp = state.queue.Pop();
+    /* If queue is not empty, pop a testcase and mutate it by ProcessInput */
+    inp = std::make_unique<QueueItem>(state.queue.Pop());
+    (*succ_nodes[1])(inp); // process_next_input
```

Comment:
Which one is a typo? And also, using an immediate value is not good this way.
I suggest you to make this routine receive these two nodes in its constructor. 
Then, you can use `idx` (maybe you need to add some getters since this value is protected).
https://github.com/fuzzuf/fuzzuf/blob/master/include/fuzzuf/hierarflow/hierarflow_callee.hpp#L99

Reply:
Fixed in [61271e6](https://github.com/fuzzuf/fuzzuf/pull/23/commits/61271e690689875e10006cc6e3a87d480bfa11f8)
This commit introduces a new method `GetCalleeIndexRef` to `HierarFlowNode` (and `HierarFlowCallee`)